src/storage/src/hummock/compactor/mod.rs

```diff
@@ -183,12 +183,17 @@ impl Compactor {
             .await;
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
-        generate_splits(&mut compact_task, context.clone()).await;
+        let mut task_status = TaskStatus::Success;
+        if let Err(e) = generate_splits(&mut compact_task, context.clone()).await {
+            tracing::warn!("Failed to generate_splits {:#?}", e);
+            task_status = TaskStatus::ExecuteFailed;
+            Self::compact_done(&mut compact_task, context.clone(), vec![], task_status).await;
+            return task_status;
```

Comment:
```rust
        context.compactor_metrics.compact_task_pending_num.dec();

```

If you choose to exit early, remember to release some metrics, perhaps we can use guard